pkg/pillar/cmd/zedrouter/zedrouter.go

```diff
@@ -2938,23 +2948,31 @@ func checkUplinkPortOverlap(ctx *zedrouterContext, network string, network1 stri
 // scan through existing AppNetworkStatus list to bring
 // up any AppNetwork struck in error state, while
 // contending for resource
-func scanAppNetworkStatusInErrorAndUpdate(ctx *zedrouterContext, key0 string) {
+func scanAppNetworkStatusInErrorAndUpdate(ctx *zedrouterContext, timerTrig bool) {
 	log.Infof("scanAppNetworkStatusInErrorAndUpdate()\n")
 	pub := ctx.pubAppNetworkStatus
 	items := pub.GetAll()
 	for key, st := range items {
 		status := cast.CastAppNetworkStatus(st)
 		config := lookupAppNetworkConfig(ctx, key)
 		if config == nil || !config.Activate ||
-			status.Error == "" || key == key0 {
+			status.Error == "" {
 			continue
 		}
 		// We wouldn't have even copied underlay/overlay
 		// networks into status. This is as good as starting
 		// from scratch all over. App num that would have been
 		// allocated will be used this time also, since the app UUID
 		// does not change.
-		handleAppNetworkCreate(ctx, key, *config)
+		if timerTrig {
+			// When the function is called from the timer, run the AppNetworkCreate to retry
+			log.Infof("scanAppNetworkStatusInErrorAndUpdate: run create\n")
+			handleAppNetworkCreate(ctx, key, *config)
+		} else {
+			// When hit error while creating, set a timer and wait for sometime and come back
+			log.Infof("scanAppNetworkStatusInErrorAndUpdate: set timer\n")
+			ctx.appNetCreateTimer = time.NewTimer(60 * time.Second)
```

Comment:
If you have two or more ANS with Error set you don't want to start the timer twice. A separate function would make this more clear.

Also, I think we should have a configItem for the timer value.

Reply:
The later ANS error will just reset the timer(which is not a big deal), when the timer fires, it will walk through all the ANS which had error before and retry them all.